src/app/modules/main/wallet_section/transactions/module.nim

```diff
@@ -136,9 +136,13 @@ method authenticateAndTransfer*(self: Module, from_addr: string, to_addr: string
   ##################################
 
 method onUserAuthenticated*(self: Module, password: string) =
-  self.controller.transfer(self.tmpSendTransactionDetails.fromAddr, self.tmpSendTransactionDetails.toAddr,
-    self.tmpSendTransactionDetails.tokenSymbol, self.tmpSendTransactionDetails.value, self.tmpSendTransactionDetails.uuid,
-    self.tmpSendTransactionDetails.priority, self.tmpSendTransactionDetails.selectedRoutes, password)
+  if password.len == 0:
+    let response = %* {"uuid": self.tmpSendTransactionDetails.uuid, "success": false, "error": "cancelled"}
+    self.view.transactionWasSent($response)
+  else:
+    self.controller.transfer(self.tmpSendTransactionDetails.fromAddr, self.tmpSendTransactionDetails.toAddr,
+      self.tmpSendTransactionDetails.tokenSymbol, self.tmpSendTransactionDetails.value, self.tmpSendTransactionDetails.uuid,
+      self.tmpSendTransactionDetails.priority, self.tmpSendTransactionDetails.selectedRoutes, password)
```

Comment:
@Khushboo-dev-cpp this won't work if user is a keycard user.

In the keycard context `password` field is actually public key of encryption keys set and it's used as it is to encrypt the database, but calling `transfer` in `backend/transactions.nim` file in `createMultiTransaction` proc we hash it in a way `"0x" & $keccak_256.digest(password)` which is incorrect. Also on the status-go side `VerifyAccountPassword` function won't pass for keycard user.

Reply:
Hey @saledjenic , i just followed what was done for AddAccount in wallet, but maybe its update by now?
Do you have an example for the flow I can follow, or should we add support for keycard in a separate task? and maybe you can do it cause you know it better?
Let me know what you think

Reply:
For that we need 2 steps of motifications:
1. on the `status-desktop` side, you need to pass the password without hashing it in case of keycard user, see [here](https://github.com/status-im/status-desktop/blob/master/src/app_service/service/wallet_account/service.nim#L266-L286), check lines `271` and `279` there. 
2. on the status-go side what I did for adding accounts is that I added new functions (the same as previous ones) but without verifying password, cause password is already verified on the client side, that's how I overcame that issue (I could introduce a flag in function like `skipPassVerification bool`, but that would affect mobile team and that's why I opted in for adding a new method with explicit naming). 

If you or anybody else could take it and handle that, would be great if not I can deal with it but not soon, cause I have a lot of other things to do before that.

Separate issue looks like a good option for this. Also calling transfer is not the only place where we should handle a keycard user a bit differently.

Reply:
@saledjenic I agree, ens names, buy stickers there are quite a few places where this update is needed.
Iv created a follow up task for this - https://github.com/status-im/status-desktop/issues/8566
probably pick it up after whatever is planned for this cycle is done.